execution_engine/src/storage/trie/mod.rs

```diff
@@ -346,6 +350,96 @@ impl ::std::fmt::Debug for PointerBlock {
     }
 }
 
+/// Represents a Merkle Tree or a chunk of data with attached proof.
+/// Chunk with attached proof is used when the requested
+/// trie is larger than [ChunkWithProof::CHUNK_SIZE_BYTES].
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum TrieOrChunkedData {
+    /// Represents a Merkle Trie.
+    Trie(Box<Trie<Key, StoredValue>>),
+    /// Represents a chunk of data with attached proof.
+    ChunkWithProof(ChunkWithProof),
+}
+
+impl Display for TrieOrChunkedData {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            TrieOrChunkedData::Trie(trie) => {
+                write!(f, "Trie({})", trie)
+            }
+            TrieOrChunkedData::ChunkWithProof(chunk) => {
+                write!(
+                    f,
+                    "ChunkWithProof {{ index = {}, hash = {} }}",
```

Comment:
```suggestion
                    "ChunkWithProof {{ index: {}, hash: {} }}",
```

Or even better: Use `Formatter::debug_struct`.